File: backend/app/schemas/update.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import UpdateRunStatus


class UpdateCheckResult(BaseModel):
    current_version: str
    latest_version: str | None
    update_available: bool
    release_url: str | None = None
    release_notes: str | None = None
    published_at: datetime | None = None
    checked_at: datetime
    error: str | None = None


class UpdateRunOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    status: UpdateRunStatus
    from_version: str
    to_version: str | None
    log: str | None
    started_at: datetime | None
    finished_at: datetime | None
    created_at: datetime

File: frontend/src/lib/update.ts
export interface UpdateCheckResult {
  current_version: string;
  latest_version: string | null;
  update_available: boolean;
  release_url: string | null;
  release_notes: string | null;
  published_at: string | null;
  checked_at: string;
  error: string | null;
}

export type UpdateRunStatus = "PENDING" | "RUNNING" | "SUCCEEDED" | "FAILED";

export const updateRunStatusLabels: Record<UpdateRunStatus, string> = {
  PENDING: "Pending",
  RUNNING: "Running",
  SUCCEEDED: "Succeeded",
  FAILED: "Failed",
};

export interface UpdateRun {
  id: string;
  status: UpdateRunStatus;
  from_version: string;
  to_version: string | null;
  log: string | null;
  started_at: string | null;
  finished_at: string | null;
  created_at: string;
}
